### ISSUE TITLE
-Wformat does not warn for inlined functions

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#18654]
|  |  |
| --- | --- |
| Bugzilla Link | [18280](https://llvm.org/bz18280) |
| Version | 3.0 |
| OS | Linux |
| Reporter | LLVM Bugzilla Contributor |
| CC | @DougGregor,@rnk |

## Extended Description 
When I give an inlined definition to a function, the warning about "non-literal format string" goes away.

For example:

---
#include <stdarg.h>
#include <stdio.h>
const char* dummy();
class Test1 {
public:
  Test1(const char* fmt, ...)  __attribute__((format(printf,2,3)));
};
class Test2 {
public:
  Test2(const char* fmt, ...)  __attribute__((format(printf,2,3)))
  { va_list a; va_start(a, fmt); vprintf(fmt, a); va_end(a); }
};

void tester() {
  Test1 t1(dummy());
  Test2 t2(dummy());
}
---

This compiles with no warning for t2 on clang ("clang++ -c test.cpp"), but produces warnings for both cases using GCC ("g++ -c test.cpp").

**Comment by wheatman** [comment]
This seems to have been fixed between 3.0.0 and 3.2
https://godbolt.org/z/b9KzEffxM

for reference

code
```

#include <stdarg.h>
#include <stdio.h>
const char* dummy();
class Test1 {
public:
  Test1(const char* fmt, ...)  __attribute__((format(printf,2,3)));
};
class Test2 {
public:
  Test2(const char* fmt, ...)  __attribute__((format(printf,2,3)))
  { va_list a; va_start(a, fmt); vprintf(fmt, a); va_end(a); }
};

void tester() {
  Test1 t1(dummy());
  Test2 t2(dummy());
}
```

warnings that were added

```
<source>:16:12: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
  Test1 t1(dummy());
           ^~~~~~~
<source>:17:12: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
  Test2 t2(dummy());
           ^~~~~~~
2 warnings generated.
Compiler returned: 0
``
